lib/logmsg/type-hinting.c

```diff
@@ -91,10 +99,13 @@ _is_value_hex(const gchar *value)
 }
 
 gboolean
-type_cast_to_int32(const gchar *value, gint32 *out, GError **error)
+type_cast_to_int32(const gchar *value, gssize value_len, gint32 *out, GError **error)
 {
   gchar *endptr;
 
+  if (value_len >= 0)
+    APPEND_ZERO(value, value, value_len);
```

Comment:
The use of `APPEND_ZERO()` here is unsafe in my opinion as it allocates memory on the stack based on `value_len`, which is unverified input (for example, a network source +`$(format-json)`).

We could implement our own `strntol`, which would not be as fast, or alternatively, we could make `APPEND_ZERO()` safer by limiting the allocation size (maybe large allocations could go to the heap).

Reply:
Hmm.. why would value_len be unverified? It's the length of a value, stored in NVTable. It is true it can be quite long, but the stack is usually 4-8MB so we should fit. On the other hand, you might be right that we could limit the allocation size, especially for int/double/boolean formats.